### PR TITLE
Handle class methods in EmptyLineBetweenDefs

### PR DESCRIPTION
**Pull request by unmanbearpig**
I've made it check `defs` node type as well as `def`, that's the
only change that was needed to make it work with class methods.

### CODE REVIEW

File: spec/rubocop/cop/style/empty_line_between_defs_spec.rb [hunk 1]
```diff
@@ -89,6 +89,38 @@
     end
   end
 
+  context 'class methods' do
+    let(:offending_source) do
```
Comment (bbatsov) [inline spec/rubocop/cop/style/empty_line_between_defs_spec.rb]: We should have a spec about the adjacent def & defs nodes.

Comment (unmanbearpig) [inline spec/rubocop/cop/style/empty_line_between_defs_spec.rb]: Good point, I haven’t thought about that. I’ll implement it shortly.

**Comment by unmanbearpig** [comment]
I've added tests for adjacent def and defs, what do you think about it?
I'm a bit concerned about code duplication in the specs, but I can't think of a way to refactor them without making them harder to read, any suggestions?

File: lib/rubocop/cop/style/empty_line_between_defs.rb [hunk 2]
```diff
@@ -21,10 +21,13 @@ def on_def(node)
           add_offense(node, :keyword)
         end
 
+        alias_method :on_defs, :on_def
```
Comment (bbatsov) [inline lib/rubocop/cop/style/empty_line_between_defs.rb]: We have a mixin called `OnMethodDef` or something like this, which you should be using when having to deal with `def` and `defs` nodes.

**Comment by unmanbearpig** [comment]
Thanks for the tip, I've changed it to use `OnMethodDef`, I think it's a bit better this way.

**Comment by bbatsov** [comment]
Yeah, looks good now. Just rebase it and we're good to go.

**Comment by unmanbearpig** [comment]
Done.

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**